webapp/dojo_news/src/components/NewsListItem.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <div>
+    <h2>{{title}} ({{votes}})</h2>
+    <button @click="$emit('upvote')">Upvote</button>
+    <button @click="$emit('downvote')">Downvote</button>
```

Comment:
```suggestion
    <button @click="$emit('update', item)">Upvote</button>
    <button @click="$emit('update', item)">Downvote</button>
```

That's why I'm suggesting to pass the entire data object to the child component and emit the updated data object to the parent. Your interface is much smaller, e.g. you only need one property and you only need one event to update the item. The parent component does not need to know so many details about the data item object leading to loose coupling.

Reply:
I am actually of the opinion that your suggestion broadens the interface and introduces tight coupling: Yes, instead of two properties only one is passed down - but instead of primitives being passed down it's now an object which is more complex by nature. When extending the definition of `item`, every extension gets exposed to the child component as well.
Additionally, passing up a singular generic event `update` broadens the "interface" facing upwards, so to speak - again, it's a complex object passed upwards, of which no data is actually needed. While being generic in the sense of DRY is often a good thing, this violates the SRP a bit too much for my taste; there's no code to be shared between `item.votes++` and `item.votes--`.

(We thought about adding TypeScript, but hold off for now. With typescript, adding an `interface` definition would make the downwards path clearer - but unfortunately, for sending events upwards it's awkward and would require a cast on both ends..)